trakt/shows.go

```diff
@@ -762,12 +762,125 @@ func WatchedShowsProgress() (shows []*ProgressShow, err error) {
 	}
 	wg.Wait()
 
+	// first approach
+	hiddenShowsMap := GetHiddenShowsMap("progress_watched")
 	for _, s := range showsList {
 		if s != nil {
-			shows = append(shows, s)
+			if !hiddenShowsMap[s.Show.IDs.Trakt] {
+				shows = append(shows, s)
+			} else {
+				log.Debugf("Will supress hidden show: %s", s.Show.Title)
+			}
+		}
+	}
+
+	// second approach
+	//shows = FilterHiddenProgressShows(showsList)
+
+	return
+}
+
+// GetHiddenShowsMap returns a map with hidden shows that can be used for filtering
+func GetHiddenShowsMap(section string) map[int]bool {
+	hiddenShowsMap := make(map[int]bool)
+	if config.Get().TraktToken == "" || !config.Get().TraktSyncHidden {
+		return hiddenShowsMap
+	}
+
+	hiddenShowsProgress, _ := ListHiddenShows(section, false)
+	for _, show := range hiddenShowsProgress {
+		hiddenShowsMap[show.Show.IDs.Trakt] = true
+	}
+
+	return hiddenShowsMap
+}
+
+// FilterHiddenProgressShows returns a slice of ProgressShow without hidden shows
+func FilterHiddenProgressShows(inShows []*ProgressShow) (outShows []*ProgressShow) {
+	hiddenShowsMap := make(map[int]bool)
```

Comment:
this should be placed after checking for Trakt account, to avoid allocation where not needed.